### PR TITLE
Changes default render threads from all to all - 1

### PR DESCRIPTION
**Pull request by Mango-3**
- Implements request #314

### CODE REVIEW

File: src/appleseed-max-impl/appleseedrenderer/renderersettings.cpp [hunk 1]
```diff
@@ -369,7 +369,7 @@ void RendererSettings::apply_settings_to_interactive_config(asr::Project& projec
     params.insert_path("lighting_engine", "pt");
 
     if (m_rendering_threads == 0)
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedrenderer/renderersettings.cpp]: I wonder if this whole block still makes sense, now that the default for m_rendering_threads is sane (-1). There's no special case for final configurations.

Comment (Mango-3) [reply src/appleseed-max-impl/appleseedrenderer/renderersettings.cpp]: For ActiveShade mode I think it still makes sense that when rendering_threads == 0 (e.g. from loading saved scene files) it is always overridden to keep the UI responsive.  
For production rendering mode it is less of an issue as one does not actively manipulate the scene during rendering (as one usually does in ActiveShade mode).

Comment (dictoon) [reply src/appleseed-max-impl/appleseedrenderer/renderersettings.cpp]: Ok.

In that case, the code that sets the number of rendering threads in `apply_common_settings()` could be moved to `apply_settings_to_final_config()`. I think that would make things a little clearer.

Comment (Mango-3) [reply src/appleseed-max-impl/appleseedrenderer/renderersettings.cpp]: Yes, absolutely right.

File: src/appleseed-max-impl/appleseedrenderer/renderersettings.cpp [hunk 2]
```diff
@@ -357,6 +353,7 @@ void RendererSettings::apply_settings_to_final_config(asr::Project& project) con
         params.insert_path("adaptive_tile_renderer.max_samples", m_adaptive_max_samples);
         params.insert_path("adaptive_tile_renderer.noise_threshold", m_adaptive_noise_threshold);
     }
+    params.insert_path("rendering_threads", m_rendering_threads);
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedrenderer/renderersettings.cpp]: You need to preserve the logic that set the parameter to `auto` when it's 0 in Max.

Comment (Mango-3) [reply src/appleseed-max-impl/appleseedrenderer/renderersettings.cpp]: Yes, definitely true.

**Pull request merged**